go/apps/jsbox/outbound.py

```diff
@@ -49,19 +49,48 @@ def is_inbound_push_trigger(msg):
     return bool(msg.get(INBOUND_PUSH_TRIGGER, False))
 
 
+class InvalidHelperMetadata(Exception):
+    """
+    Internal exception raised when a sandboxed application
+    sends invalid helper_metadata.
+    """
+
+
 class GoOutboundResource(SandboxResource):
     """Resource that provides outbound message support for Go.
 
     Includes support for replying, replying to groups and sending
     messages via given tags.
+
+    Configuration options:
+
+    :param list allowed_helper_metadata:
+        List of helper_metadata fields that may be set by sandboxed
+        applications.
     """
 
+    def setup(self):
+        self._allowed_helper_metadata = set(
+            self.config.get('allowed_helper_metadata', []))
+
     def _mkfail(self, command, reason):
         return self.reply(command, success=False, reason=reason)
 
     def _mkfaild(self, command, reason):
         return succeed(self._mkfail(command, reason))
 
+    def _check_helper_metadata(self, helper_metadata):
+        if any(key not in self._allowed_helper_metadata
+               for key in helper_metadata.iterkeys()):
+            if self._allowed_helper_metadata:
+                errmsg = (
+                    "'helper_metadata' may only contain the following keys: %s"
+                    % ', '.join(sorted(self._allowed_helper_metadata)))
+            else:
+                errmsg = (
+                    "'helper_metadata' is not allowed")
+            raise InvalidHelperMetadata(errmsg)
```

Comment:
I found this logic flow a little tricky. Would it make sense to rewrite it along the following lines?
```python
if not helper_metadata:
    return  # Nothing to check
if not self._allowed_helper_metadata:
    raise InvalidHelperMetadata("...")
if any(...):
    raise InvalidHelperMetadata("...")
```